staging/src/k8s.io/code-generator/cmd/client-gen/generators/generator_for_type.go

```diff
@@ -459,6 +461,13 @@ func new$.type|publicPlural$(c *$.GroupGoName$$.Version$Client) *$.type|privateP
 var listTemplate = `
 // List takes label and field selectors, and returns the list of $.resultType|publicPlural$ that match those selectors.
 func (c *$.type|privatePlural$) List(ctx context.Context, opts $.ListOptions|raw$) (result *$.resultType|raw$List, err error) {
+    return c.list(ctx, opts)
+}
+`
+
+var privateListTemplate = `
```

Comment:
I'm not sure I understand the reasoning behind this change.

Reply:
it allows for passing `c.list` to the `cache.CheckListAgainstCacheDataConsistencyIfRequested` :

```
cache.CheckListAgainstCacheDataConsistencyIfRequested(ctx, "list request for examples", c.list, opts, result)
```